src/pages/settings/Profile/PronounsPage.js

```diff
@@ -58,60 +48,38 @@ function PronounsPage(props) {
             .sortBy((pronoun) => pronoun.text.toLowerCase())
             .value();
 
-        setPronounsList(pronouns);
+        const trimmedSearch = searchValue.trim();
 
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [props.currentUserPersonalDetails.pronouns]);
+        if (trimmedSearch.length === 0) {
+            return [];
+        }
+        return _.filter(pronouns, (pronoun) => pronoun.text.toLowerCase().indexOf(trimmedSearch.toLowerCase()) >= 0);
+    }, [searchValue, currentPronouns, translate]);
 
-    const onChangeText = (value = '') => {
-        setSearchValue(value);
-    };
+    const headerMessage = searchValue.trim() && !filteredPronounsList.length ? translate('common.noResultsFound') : '';
```

Comment:
I see you did more strict condition above, so we can do it here as well
```suggestion
    const headerMessage = searchValue.trim() && filteredPronounsList.length === 0 ? translate('common.noResultsFound') : '';
```

Reply:
Done!